### ISSUE TITLE
DB Problem writing a comment to an ebook

### ISSUE THREAD

**Issue by n0valis** [janeczku/calibre-web#2278]
**Describe the bug/problem**
It is probably not an issue with calibreweb but with the ebbok but I would like to understand what happened and why.
If I update a description of a specific ebook I get the following log message:

/opt/calibre-web-master/vendor/sqlalchemy/orm/session.py:1911: SAWarning: Identity map already had an identity for (<class 'cps.db.Comments'>, (529,), None), replacing it with newly flushed object.   Are there load operations occurring inside of an event handler within the flush?

After that the description of the ebook is present and ok in the metadata.db. If I check with the Calibre main programm I can read the description in the metadata.

In calibreweb though the description nis not present nor can I edit the book again. The pencil icon disappeard and that's it.
I can still download the book from the download icon.

So what could be the reasons in calibreweb for not showing that edit icon?

 - OS: DietPI  v7.8.2 on Raspi4 headless
 - Python version: Python 3.9.2
 - Calibre-Web version: 0.6.8 Beta

Thank you for your help.

**Comment by OzzieIsaacs** [comment]
The error message sounds to me there is another comments entry linked to the book. 
Please perform a library maintenance in calibre itself (or check for the double entry in comments table in metadata.db). 
I have no idea how this could happen. After a restart of calibre-web editing should work again.

The only reason for the edit icon to not show up is if the user is missing edit rights:
https://github.com/janeczku/calibre-web/blob/35209ede67622eeea4079e55bd41f3f1c06da7e7/cps/templates/detail.html#L295

**Comment by n0valis** [comment]
I don't know, I don't think there is a description/comment already present. At least I couldn't find one.
Also the edit function will not return after restarting calibreweb.
If it is indeed a rights issue, it then must have been caused by calibreweb itself, as it has the capability to write/edit the file immediately after import. But not after editing the description. Editing anything else doesn't affect the edit status.
I can provide the file if interested.

**Comment by n0valis** [comment]
Here is the link 
[Ebook](https://drive.google.com/file/d/1PmfHTG29sNi7WFl2nwkjOItNF_yCFiq1/view?usp=drivesdk)

**Comment by OzzieIsaacs** [comment]
I uploaded the book to calibre-web added a comment ("h"), saved it, edited it again. No problem
I added the book in calibre (version 5.34), added a comment in calibre-web, saved it. edited it again. No problem
I edited the comment in calibre saved it. Refreshed the calibre-web page.  No problem
I opened the edit book page in calibre-web, in parallel I edited the comment in calibre, saved it there. Changed the comment without refresh in calibre-web, saved it, no problem. 
Changed the comment in calibre-web, saved it. I edited the comment afterwards in calibre (without restart of calibre like I also did before), saved it there. Refreshed page in calibre-web no problem (comment from calibre is visible). 
The problem is not the book it's the calibre database (metadata.db).
One remark: Calibre-web doesn't edit the file itself, it only edits the metadata.db file.
